WordPressUI/Extensions/UIImageView+Gravatar.swift

```diff
@@ -51,12 +51,32 @@ extension UIImageView {
     ///     - placeholderImage: Image to be used as Placeholder
     ///
     @objc
-    public func downloadGravatarWithEmail(_ email: String, rating: GravatarRatings = .`default`, placeholderImage: UIImage = .gravatarPlaceholderImage) {
-        let gravatarURL = gravatarUrl(for: email, size: gravatarDefaultSize(), rating: rating.stringValue())
+    public func downloadGravatarWithEmail(_ email: String, rating: GravatarRatings = .default, placeholderImage: UIImage = .gravatarPlaceholderImage) {
+        let gravatarURL = gravatarUrl(for: email, size: gravatarDefaultSize(), rating: rating)
 
+        listenForGravatarChanges()
         downloadImage(from: gravatarURL, placeholderImage: placeholderImage)
     }
 
+    private func listenForGravatarChanges() {
+        guard downloadURL == nil else {
+            return
+        }
+
+        NotificationCenter.default.addObserver(forName: .GravatarImageUpdateNotification, object: nil, queue: nil) { [weak self] (notification) in
+            guard let userInfo = notification.userInfo,
+                let email = userInfo["email"] as? String,
+                let image = userInfo["image"] as? UIImage,
+                let downloadURL = self?.downloadURL else {
+                return
+            }
+            let testHash = self?.gravatarHash(of: email) ?? ""
+            if downloadURL.absoluteString.contains(testHash) {
+                self?.image = image
+            }
+        }
+    }
```

Comment:
I'm afraid this might not work. I've recently done something similar, and bottom line is:

- This method returns an opaque object
- As per the docs, you're supposed to keep it around, and [use it to remove the observer](https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver)
- In my actual tests (in another PR!), if you don't retain it, the `observer` block gets removed automatically

Plus: There is no "remove observer" mechanism.

Reply:
Correction: The "dealloc gets me unsubscribed" was actually KVO. But still, we could use a "remove observer" call, otherwise the block will end up lurking indefinitely.

Reply:
it definitely _is_ working, but I think you're right—there needs to be a remove call somewhere. 👍